libs/datamodel/core/src/transform/ast_to_dml/db/relations.rs

```diff
@@ -0,0 +1,219 @@
+use super::{context::Context, types::RelationField};
+use crate::ast;
+use std::collections::BTreeSet;
+
+#[derive(PartialOrd, Ord, PartialEq, Eq, Debug, Clone)]
+enum OneToManyRelationFields {
+    Forward(ast::FieldId),
+    Back(ast::FieldId),
+    Both(ast::FieldId, ast::FieldId),
+}
+
+#[derive(PartialOrd, Ord, PartialEq, Eq, Debug, Clone)]
+enum RelationType {
+    ImplicitManyToMany {
+        field_a: ast::FieldId,
+        field_b: ast::FieldId,
+    },
+    OneToOne {
+        field_a: ast::FieldId,
+        field_b: ast::FieldId,
+    },
+    OneToMany(OneToManyRelationFields),
+}
+
+#[derive(PartialOrd, Ord, PartialEq, Eq, Debug, Clone)]
+pub(super) struct Relation<'ast> {
+    /// The `name` argument in `@relation`.
+    relation_name: Option<&'ast str>,
+    tpe: RelationType,
+}
+
+#[derive(Debug, Default)]
+pub(super) struct Relations<'ast> {
+    // A relation is always between two models. One model is assigned the role
```

Comment:
This should be rustdoc even when we don't have a public struct. I think you can still generate docs even from private types, that could be useful e.g. on github pages.

Reply:
So I opted against that because rustdocs are _api docs_ to me, not implementation docs. But I'd be ok making some of these rustdocs (the ones that concern specific fields).

Reply:
We should maybe have a searchable rustdoc rendering available from datamodel, WDYT?

Reply:
For all of prisma-engines, yes.